dags/synapse-dataset-to-croissant.py

```diff
@@ -676,8 +669,11 @@ def execute_push_to_synapse(push_to_synapse: bool, dataset: Entity, dataset_id:
             _rest_call_replacement, authenticated_syn_client)
         existing_row = authenticated_syn_client.tableQuery(
             query=f"SELECT * FROM {SYNAPSE_TABLE_FOR_CROISSANT_LINKS} WHERE dataset = '{dataset_id}' AND dataset_version = {dataset_version} AND dataset_collection = '{dataset_collection}'", resultsAs="csv")
-        existing_row_df = existing_row.asDataFrame()
-        os.remove(existing_row.filepath)
+
```

Comment:
hmm I am curious why this got turned into a try except block? is it because existing_row can be None?

Reply:
This is to make sure that if an exception is raised during `existing_row.asDataFrame()` that the file is always removed.

`existing_row` wouldn't be `None`, but, it could error inside the function.